language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/helper/ListHelper.java

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.javaapi.data.helper;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public final class ListHelper {
+    private ListHelper() {}
+    public static <T> List<T> list(T... args) {
+        List<T> ret = new ArrayList<>();
+        for(T arg : args){
+            ret.add(arg);
+        }
+        return ret;
+    }
+}
```

Comment:
Would [this](https://docs.oracle.com/javase/8/docs/api/java/util/Arrays.html#asList-T...-) work for you? Starting from Java 9 we can also use [this](https://docs.oracle.com/javase/9/docs/api/java/util/List.html#immutable).

Reply:
I'm aware of both, but we're stuck on 1.8 and I think
```
list(party), list(),
```
reads a lot better than
```
Arrays.toList(new String[]{party}), Arrays.toList(new String[]{}),
```

Somehow it never occurred to me before right now to use `Arrays.toList` in the implementation of `list`, though. On the off chance that is what you were suggesting, I've done that now.

Reply:
Are you sure that is how the syntax would work? The documentation itself says:

> This method also provides a convenient way to create a fixed-size list initialized to contain several elements:

```java
Arrays.asList("Larry", "Moe", "Curly");
```

Reply:
By the way, since we're here, in the future you may also find [Collections.singletonList](https://docs.oracle.com/javase/8/docs/api/java/util/Collections.html#singletonList-T-) and [Collections.emptyList](https://docs.oracle.com/javase/8/docs/api/java/util/Collections.html#emptyList--) useful.